rules/prefer-math-trunc.js

```diff
@@ -41,26 +30,41 @@ const bitwiseNotUnaryExpressionSelector = [
 
 const create = context => {
 	const sourceCode = context.getSourceCode();
+
 	const mathTruncFunctionCall = node => {
 		const text = sourceCode.getText(node);
 		const parenthesized = node.type === 'SequenceExpression' ? `(${text})` : text;
 		return `Math.trunc(${parenthesized})`;
 	};
 
 	return {
-		[bitwiseOrBinaryExpressionSelector]: node => {
-			context.report({
-				node,
-				messageId: MESSAGE_ID_BITWISE_OR,
-				fix: fixer => fixer.replaceText(node, mathTruncFunctionCall(node.left))
-			});
+		'BinaryExpression[right.type="Literal"]': node => {
+			const {operator, right} = node;
+			if (binaryOperators.has(operator) && right.value === 0) {
```

Comment:
We are now using  `value === 0`, we need add some tests to cover other `0`

```js
foo | .0;
foo | 0.;
foo | 0.00_00;
foo | 0b0;
foo | 0x0000_0000_0000;
```

Please add

```js
foo | (0.4 + .6);
```

as well

Reply:
Uh yes, just wondering, how can I make the math expressions work?? With an eval? Making the computations directly in the code? Or does ESLint have a util function to do that?

Reply:
No, we don't need make it work.

Reply:
In [this rule](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/859/files#diff-a1cd7e3eacfd287d9522bdaa755f05111c27b9de74197392c5ceb4b6b68a2762R40) there're several cases equals to `1`, but we don't want it work.

Reply:
Oh you want me to add them to the `valid` case?

Reply:
Yes

Reply:
Add `foo & 0xFFFFFFFFFFFF` too, we do't want other people think about `&` again.